encointer/encointer-parachain
# Enable Fee Payment in community currency

last feature addition before 1.0.0 release
* bump polkadot deps to 0.9.18
* bump pallets to 1.0.0
* define existential deposit for community currency
* enable asset fee payment for community currency

Tested with kusama-local
* [x] bootstrap_demo_community :+1: 
* [x] pay fees in CC, even if no native token on account
* [x] bot-community growth :+1: 

Test-Setup:
* Use api-client-hotfix branch: https://github.com/scs/substrate-api-client/tree/cl/encointer-community-currency-fees-hotfix.
   * prepare binary `encointer-client-notee-ccfee` with `cargo update -p substrate-api-client --precise ca8a6cf7d18362affbadc7380382f9f981058964`
   * prepare binary `encointer-client-notee` with `cargo update -p substrate-api-client --precise b51d802b054bbd655b3178448822e2db04a8a71e`
* Set Demurrage to 0.
Tested:
1. run `bootstrap_demo_community.py` script using the native token.
2. Verify: `./encointer-client-notee balance --cid sqm1v79dF6b //Alice` == 1.0
3. Rebuild api-client to use hardcodedly the community currency as fee.
4. Send extrinsic with Alice: `./encointer-client-notee next-phase`
5. Verify: `./encointer-client-notee balance --cid sqm1v79dF6b //Alice` < 1.0

Result: Community fee was successfully withdrawn, which is many orders of magnitude bigger than for the solo-node:

Logs:
```
 ./bootstrap_demo_community.py                                                       

...

Balances for new community with cid: sqm1v79dF6b.
Account balance for //Alice is 1.0.
Account balance for //Bob is 1.0.
Account balance for //Charlie is 1.0.
[('1', ' sqm1v79dF6b', 'VerifiedUnlinked')]
tests passed

./encointer-client-notee balance --cid sqm1v79dF6b //Alice 
1
./encointer-client-notee next-phase
Phase is now: ASSIGNING
 ./encointer-client-notee balance --cid sqm1v79dF6b //Alice
0.9993701999
```

test of account creation without native tokens:
```
./bootstrap_demo_community.py 
../target/release/encointer-client-notee transfer //Alice //Ferdie 0.5 --cid sqm1v79dF6b 
../target/release/encointer-client-notee  balance //Ferdie
../target/release/encointer-client-notee  balance --cid sqm1v79dF6b //Ferdie
../target/release/encointer-client-notee-ccfee transfer //Ferdie //Eve 0.1 --cid sqm1v79dF6b 
../target/release/encointer-client-notee  balance --cid sqm1v79dF6b //Eve
../target/release/encointer-client-notee  balance --cid sqm1v79dF6b //Ferdie
```

test community growth (classic, as usual)
```
./bot-community.py init
./bot-community.py test
```
